spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/AbstractStreamService.java

```diff
@@ -121,25 +111,7 @@ else if (DeploymentState.deploying.equals(DeploymentState.valueOf(status))) {
 	// State
 	@Override
 	public Map<StreamDefinition, DeploymentState> state(List<StreamDefinition> streamDefinitions) {
-		Map<StreamDefinition, DeploymentState> states = new HashMap<>();
-		List<StreamDefinition> deployerSpecificStreamDefinitions = new ArrayList<>();
-		for (StreamDefinition streamDefinition : streamDefinitions) {
-			StreamDeployment streamDeployment = this.streamDeploymentRepository.findOne(streamDefinition.getName());
-			if (streamDeployment == null) {
-				states.put(streamDefinition, DeploymentState.unknown);
-			}
-			else if (this.streamDeployer == StreamDeployers.valueOf(streamDeployment.getDeployerName())) {
-				deployerSpecificStreamDefinitions.add(streamDefinition);
-			}
-			else {
-				logger.error("Invalid deployer:" + streamDefinition.getName() + ":" + streamDeployment.getDeployerName());
-				throw new IncompatibleStreamDeployerException(this.streamDeployer.name());
-			}
-		}
-		if (!deployerSpecificStreamDefinitions.isEmpty()) {
-			states.putAll(this.doState(deployerSpecificStreamDefinitions));
-		}
-		return states;
+		return this.doState(streamDefinitions);
```

Comment:
similar as the doUndeployStream we can remove the redundant doState and use the state method instead.

Reply:
+1

Reply:
we can remove the `doState` and use directly the `state` method instead